csrc/fusion_executor/executor.cpp

```diff
@@ -1577,6 +1577,19 @@ void FusionExecutor::initializeExecutorEntry(
 
   executor_utils::validateCircularBuffering(kernel(), expr_eval);
 
+  // Check that a full warp exists in blockDim.x if the kernel contains
+  // ElectSync predicate.
+  bool has_elect_sync_predicate =
```

Comment:
I am not very familiar with `FusionExecutor`, but my understanding is `initializeExecutorEntry` will only be invoked at the first time when we run a fusion. Which means, what could happen is, the first time when we run a fusion, a good BIDx value is provided. But later, we get a bad BIDx, but we will not run this check again. Should we make `has_elect_sync_predicate` a kernel summary (so that we don't have to run this full fusion traversal on `fusion()->vals()` over and over again), and move this check to `runFusion`?

Reply:
Note that `fusion()->vals()` is fast as it just returns a reference to the unordered_set of all vals registered in the `IrContainer`; it doesn't do a traversal or topo sort like `usedMathVals()` does. That also means it looks for an `ElectSync` predicate even if it's no longer "alive" in the fusion. See #2949.

Reply:
Wouldn't BDimx only change if different input sizes were given? I believe in this case, we call `initializeExecutorEntry` again.

Reply:
`initializeExecutorEntry` is called whenever we see a new set of inputs.
Reference: https://github.com/NVIDIA/Fuser/blob/main/csrc/fusion_executor/executor.cpp#L1220-L1234

Reply:
I'll move `has_elect_sync_predicate` to the kernel summary.